### PR TITLE
Protractor tests

### PR DESCRIPTION
**Pull request by garethbowen**

### CODE REVIEW

File: tests/protractor/conf.js [hunk 1]
```diff
@@ -70,7 +70,7 @@ const setupUser = () => {
 
 exports.config = {
   seleniumAddress: 'http://localhost:4444/wd/hub',
-  specs: [ 'e2e/**/*.js' ],
+  specs: [ 'e2e/**/add-user.specs.js' ],
```
Comment (garethbowen) [inline tests/protractor/conf.js]: Revert this back to *.js

File: tests/protractor/conf.js [hunk 2]
```diff
@@ -88,5 +88,5 @@ exports.config = {
     browser.driver.sleep(1000);
     return login(browser);
   },
-  onCleanUp: () => modules.forEach(module => module.kill())
+ onCleanUp: () => modules.forEach(module => module.kill())
```
Comment (garethbowen) [inline tests/protractor/conf.js]: Revert this back too.

File: tests/protractor/helper.js [hunk 3]
```diff
@@ -1,6 +1,12 @@
-
+var fs = require('fs');
 var EC = protractor.ExpectedConditions;
 
+ function writeScreenShot(data, filename) {
+        var stream = fs.createWriteStream(filename);
+        stream.write(new Buffer(data, 'base64'));
+        stream.end();
+    }
```
Comment (garethbowen) [inline tests/protractor/helper.js]: Fix indentation.

File: tests/protractor/page-objects/login/login.po.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+var helper = require('../../helper');
+
+var incorrectCredentialsText = 'Incorrect user name or password. Please try again.';
+var passwordBlankText = 'Can\'t be blank.';
+var usernameField = element(by.id('user'));
+var passwordField = element(by.id('password'));
+var loginButton = element(by.id('login'));
+var incorrectCredeantialsError = element(by.className('error incorrect'));
```
Comment (garethbowen) [inline tests/protractor/page-objects/login/login.po.js]: Pull these 4 into functions to have less state.

File: tests/protractor/page-objects/users/add-user-modal.po.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+var emailField = element(by.id('email'));
+var phoneField = element(by.id('phone'));
+var languageField = element(by.id('language'));
+var userTypeField = element(by.id('type'));
+//var placeField = element(by.id('facility'));
+//var contactField = element(by.id('contact'));
+var passwordField = element(by.id('password'));
+var confirmPasswordField = element(by.id('password-confirm'));
+var submitButton = element(by.className('btn submit btn-primary'));
+var cancelButton = element(by.className('btn cancel'));
```
Comment (garethbowen) [inline tests/protractor/page-objects/users/add-user-modal.po.js]: Pull all these into functions to have less state.

File: tests/protractor/page-objects/users/add-user-modal.po.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
+var helper = require('../../helper');
+
+var usernameField = element(by.id('name'));
+var fullNameField = element(by.id('fullname'));
+var emailField = element(by.id('email'));
+var phoneField = element(by.id('phone'));
+var languageField = element(by.id('language'));
+var userTypeField = element(by.id('type'));
+//var placeField = element(by.id('facility'));
+//var contactField = element(by.id('contact'));
```
Comment (garethbowen) [inline tests/protractor/page-objects/users/add-user-modal.po.js]: Remove commented out code.

File: tests/protractor/page-objects/users/users.po.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
+var helper = require('../../helper');
+var addUserButton = element(by.id('add-user'));
+//var addUserModal = $('mm-modal#edit-user-profile');
```
Comment (garethbowen) [inline tests/protractor/page-objects/users/users.po.js]: Remove commented out code.

File: tests/protractor/page-objects/users/users.po.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+var usersList = element.all(by.repeater('user in users'));
```
Comment (garethbowen) [inline tests/protractor/page-objects/users/users.po.js]: Pull all these into functions to have less state.

File: tests/protractor/page-objects/users/users.po.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+	openAddUserModal: function () {
+		helper.waitElementToBeClickable(addUserButton);
+		addUserButton.click();
+	},
+	getUsersList:function(){
+		helper.waitUntilReady(addUserButton);
+		return usersList;
+	}
+};
```
Comment (garethbowen) [inline tests/protractor/page-objects/users/users.po.js]: The indentation in this file is all wrong. Make sure you editor is set to 2 spaces.

File: tests/protractor/utils.js [hunk 10]
```diff
@@ -175,6 +175,10 @@ module.exports = {
     `http://${auth.user}:${auth.pass}@${constants.COUCH_HOST}:${constants.COUCH_PORT}/${constants.DB_NAME}`,
 
   getBaseUrl: () =>
-    `http://${constants.API_HOST}:${constants.API_PORT}/${constants.DB_NAME}`
+    `http://${constants.API_HOST}:${constants.API_PORT}/${constants.DB_NAME}`,
+    
+getLoginUrl: () =>
+ `http://${constants.API_HOST}:${constants.API_PORT}/${constants.DB_NAME}/login`
```
Comment (garethbowen) [inline tests/protractor/utils.js]: Indentation again.

File: tests/protractor/page-objects/login/login.po.js [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+  },
+
+  getIncorrectCredentialsText: () => {
+    return incorrectCredentialsText;
+  },
+
+  getPasswordBlankText: () => {
+    return passwordBlankText;
+  }
+};
```
Comment (garethbowen) [inline tests/protractor/page-objects/login/login.po.js]: Nice use of ES6! Loving the `const`!

File: tests/protractor/page-objects/users/users.po.js [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+		helper.waitElementToBeClickable(getAddUserButton());
+		getAddUserButton().click();
+	},
+	getUsersList: function () {
+		helper.waitUntilReady(getAddUserButton());
+		return element.all(by.repeater('user in users'));
+	}
+};
+
+
```
Comment (garethbowen) [inline tests/protractor/page-objects/users/users.po.js]: This file is still indented incorrectly. It should be 2 spaces.

File: tests/protractor/utils.js [hunk 13]
```diff
@@ -54,57 +54,57 @@ var request = function(options, debug) {
...
   request: request,
 
-  requestOnTestDb: function(options, debug) {
+  requestOnTestDb: function (options, debug) {
     options.path = '/' + constants.DB_NAME + options.path;
     return request(options, debug);
   },
 
-  saveDoc: function(doc) {
+  saveDoc: function (doc) {
```
Comment (garethbowen) [inline tests/protractor/utils.js]: It looks like your formatter is configured to add spaces after the `function` keyword which isn't consistent with how our code is usually formatted. Can you try to configure your formatter to be consistent with our existing code?

**Comment by ngaruko** [comment]
Added 'login' and 'add user' tests.

**Pull request merged**